plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestPostgreSqlConnectorTest.java

```diff
@@ -836,6 +836,13 @@ public void testNotExpressionPushdown()
         }
     }
 
+    @Test
+    public void testCastPushdown()
+    {
+        assertThat(query("SELECT nationkey FROM nation WHERE CAST(nationkey AS varchar) = '1'"))
```

Comment:
Such CAST could be unwrapped by an optimization in the engine into `nationkey = 1`.
Can you think of a CAST that cannot be removed from the plan?